From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Aikar <aikar@aikar.co>
Date: Sun, 4 Dec 2022 12:48:52 -0300
Subject: [PATCH] Disable Vanilla Chunk GC

Bukkit has its own system for this.

diff --git a/src/main/java/net/minecraft/server/WorldServer.java b/src/main/java/net/minecraft/server/WorldServer.java
index 9bf5b0a7d4b7eac5496768b3ce8dad22b02b332c..95fa99a0f6629622871934f526081bfe9b54a209 100644
--- a/src/main/java/net/minecraft/server/WorldServer.java
+++ b/src/main/java/net/minecraft/server/WorldServer.java
@@ -963,6 +963,8 @@ public class WorldServer extends World implements IAsyncTaskHandler {
             this.chunkProvider.saveChunks(flag, iprogressupdate);
             } // PandaSpigot
             // CraftBukkit - ArrayList -> Collection
+            // PandaSpigot start - Disable Vanilla Chunk GC
+            /*
             Collection arraylist = this.chunkProviderServer.a();
             Iterator iterator = arraylist.iterator();
 
@@ -973,6 +975,8 @@ public class WorldServer extends World implements IAsyncTaskHandler {
                     this.chunkProviderServer.queueUnload(chunk.locX, chunk.locZ);
                 }
             }
+            */
+            // PandaSpigot end
 
         }
     }
