From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: MasterDash5 <constant4337@molecularmail.com>
Date: Mon, 13 Oct 2025 11:20:33 -0600
Subject: [PATCH] Increase chat limit to 256


diff --git a/src/main/java/net/minecraft/server/PacketPlayInChat.java b/src/main/java/net/minecraft/server/PacketPlayInChat.java
index 18358b4324a7da15d932c11f19d657345da619af..322bccf3099ce4c19bc4650b46f1a37f461d0b5b 100644
--- a/src/main/java/net/minecraft/server/PacketPlayInChat.java
+++ b/src/main/java/net/minecraft/server/PacketPlayInChat.java
@@ -9,15 +9,17 @@ public class PacketPlayInChat implements Packet<PacketListenerPlayIn> {
     public PacketPlayInChat() {}
 
     public PacketPlayInChat(String s) {
-        if (s.length() > 100) {
-            s = s.substring(0, 100);
+        // PandaSpigot start - Increase chat limit to 256
+        if (s.length() > 256) {
+            s = s.substring(0, 256);
+        // PandaSpigot end
         }
 
         this.a = s;
     }
 
     public void a(PacketDataSerializer packetdataserializer) throws IOException {
-        this.a = packetdataserializer.c(100);
+        this.a = packetdataserializer.c(256); // PandaSpigot - Increase chat limit to 256
     }
 
     public void b(PacketDataSerializer packetdataserializer) throws IOException {
