From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: MasterDash5 <constant4337@molecularmail.com>
Date: Wed, 18 Feb 2026 08:51:50 -0700
Subject: [PATCH] use valueOf in DataWatcher


diff --git a/src/main/java/net/minecraft/server/EntityBat.java b/src/main/java/net/minecraft/server/EntityBat.java
index abe696e5519f450de0c2e4e1b10ba776608510c3..dccb1dc0881a292a87fe84b1e80785e5362ea1c4 100644
--- a/src/main/java/net/minecraft/server/EntityBat.java
+++ b/src/main/java/net/minecraft/server/EntityBat.java
@@ -15,7 +15,7 @@ public class EntityBat extends EntityAmbient {
     @Override
     protected void h() {
         super.h();
-        this.datawatcher.a(16, new Byte((byte) 0));
+        this.datawatcher.a(16, Byte.valueOf((byte) 0)); // PandaSpigot - use valueOf()
     }
 
     @Override
diff --git a/src/main/java/net/minecraft/server/EntityBlaze.java b/src/main/java/net/minecraft/server/EntityBlaze.java
index 720b0d227520e56e2c30cf460c01849a4f0c13c1..9838e115fcd99c79117111d19197bef6550fd396 100644
--- a/src/main/java/net/minecraft/server/EntityBlaze.java
+++ b/src/main/java/net/minecraft/server/EntityBlaze.java
@@ -29,7 +29,7 @@ public class EntityBlaze extends EntityMonster {
     @Override
     protected void h() {
         super.h();
-        this.datawatcher.a(16, new Byte((byte) 0));
+        this.datawatcher.a(16, Byte.valueOf((byte) 0)); // PandaSpigot - use valueOf()
     }
 
     @Override
diff --git a/src/main/java/net/minecraft/server/EntityBoat.java b/src/main/java/net/minecraft/server/EntityBoat.java
index 0e6f8c03501dd6ce5bda003ae54729763201ebd3..4d6f2d1d9b20cd9bb05a0a97325cda795cb5a64a 100644
--- a/src/main/java/net/minecraft/server/EntityBoat.java
+++ b/src/main/java/net/minecraft/server/EntityBoat.java
@@ -57,9 +57,11 @@ public class EntityBoat extends Entity {
     }
 
     protected void h() {
-        this.datawatcher.a(17, new Integer(0));
-        this.datawatcher.a(18, new Integer(1));
-        this.datawatcher.a(19, new Float(0.0F));
+        // PandaSpigot start - use valueOf()
+        this.datawatcher.a(17, Integer.valueOf(0));
+        this.datawatcher.a(18, Integer.valueOf(1));
+        this.datawatcher.a(19, Float.valueOf(0.0F));
+        // PandaSpigot end
     }
 
     public AxisAlignedBB j(Entity entity) {
diff --git a/src/main/java/net/minecraft/server/EntityEnderman.java b/src/main/java/net/minecraft/server/EntityEnderman.java
index 6d5fec85c8a79df2ea299c31196bab58f722fa9e..35a46b4f501f055038be8c0c9916124fe10b20d9 100644
--- a/src/main/java/net/minecraft/server/EntityEnderman.java
+++ b/src/main/java/net/minecraft/server/EntityEnderman.java
@@ -55,9 +55,11 @@ public class EntityEnderman extends EntityMonster {
 
     protected void h() {
         super.h();
-        this.datawatcher.a(16, new Short((short) 0));
-        this.datawatcher.a(17, new Byte((byte) 0));
-        this.datawatcher.a(18, new Byte((byte) 0));
+        // PandaSpigot start - use valueOf()
+        this.datawatcher.a(16, Short.valueOf((short) 0));
+        this.datawatcher.a(17, Byte.valueOf((byte) 0));
+        this.datawatcher.a(18, Byte.valueOf((byte) 0));
+        // PandaSpigot end
     }
 
     public void b(NBTTagCompound nbttagcompound) {
diff --git a/src/main/java/net/minecraft/server/EntityMinecartAbstract.java b/src/main/java/net/minecraft/server/EntityMinecartAbstract.java
index 4bf790cdffdbc8950450644a813d9c7bb539793d..c187c52414a0f8dba33a614b433b148b39718c98 100644
--- a/src/main/java/net/minecraft/server/EntityMinecartAbstract.java
+++ b/src/main/java/net/minecraft/server/EntityMinecartAbstract.java
@@ -72,11 +72,13 @@ public abstract class EntityMinecartAbstract extends Entity implements INamableT
     }
 
     protected void h() {
-        this.datawatcher.a(17, new Integer(0));
-        this.datawatcher.a(18, new Integer(1));
-        this.datawatcher.a(19, new Float(0.0F));
-        this.datawatcher.a(20, new Integer(0));
-        this.datawatcher.a(21, new Integer(6));
+        // PandaSpigot start - use valueOf()
+        this.datawatcher.a(17, Integer.valueOf(0));
+        this.datawatcher.a(18, Integer.valueOf(1));
+        this.datawatcher.a(19, Float.valueOf(0.0F));
+        this.datawatcher.a(20, Integer.valueOf(0));
+        this.datawatcher.a(21, Integer.valueOf(6));
+        // PandaSpigot end
         this.datawatcher.a(22, Byte.valueOf((byte) 0));
     }
 
diff --git a/src/main/java/net/minecraft/server/EntityMinecartFurnace.java b/src/main/java/net/minecraft/server/EntityMinecartFurnace.java
index 908620d41116f7d0474b11ec823acf8222bebe80..9eeed5e0d461d162eef6b69f1446d4e0e1394c5a 100644
--- a/src/main/java/net/minecraft/server/EntityMinecartFurnace.java
+++ b/src/main/java/net/minecraft/server/EntityMinecartFurnace.java
@@ -20,7 +20,7 @@ public class EntityMinecartFurnace extends EntityMinecartAbstract {
 
     protected void h() {
         super.h();
-        this.datawatcher.a(16, new Byte((byte) 0));
+        this.datawatcher.a(16, Byte.valueOf((byte) 0)); // PandaSpigot - use valueOf()
     }
 
     public void t_() {
diff --git a/src/main/java/net/minecraft/server/EntitySheep.java b/src/main/java/net/minecraft/server/EntitySheep.java
index 29611a194b8a582d57159231091c66adb7c6944f..d15d29e03f0af9910235e547d6b07015e579764d 100644
--- a/src/main/java/net/minecraft/server/EntitySheep.java
+++ b/src/main/java/net/minecraft/server/EntitySheep.java
@@ -71,7 +71,7 @@ public class EntitySheep extends EntityAnimal {
 
     protected void h() {
         super.h();
-        this.datawatcher.a(16, new Byte((byte) 0));
+        this.datawatcher.a(16, Byte.valueOf((byte) 0)); // PandaSpigot - use valueOf()
     }
 
     protected void dropDeathLoot(boolean flag, int i) {
diff --git a/src/main/java/net/minecraft/server/EntitySkeleton.java b/src/main/java/net/minecraft/server/EntitySkeleton.java
index 46bff95d8bf73e34eb8e0a9b94fb9fa9e9cc5cfb..c597a271045a2dea2326abcfe7819ce69410d0c6 100644
--- a/src/main/java/net/minecraft/server/EntitySkeleton.java
+++ b/src/main/java/net/minecraft/server/EntitySkeleton.java
@@ -34,7 +34,7 @@ public class EntitySkeleton extends EntityMonster implements IRangedEntity {
 
     protected void h() {
         super.h();
-        this.datawatcher.a(13, new Byte((byte) 0));
+        this.datawatcher.a(13, Byte.valueOf((byte) 0)); // PandaSpigot - use valueOf()
     }
 
     protected String z() {
diff --git a/src/main/java/net/minecraft/server/EntitySpider.java b/src/main/java/net/minecraft/server/EntitySpider.java
index 68a253c9698d2b239e400d7a9406c4430072cc33..03c18bd25d3409779114fe9cc3ca9e473c38f786 100644
--- a/src/main/java/net/minecraft/server/EntitySpider.java
+++ b/src/main/java/net/minecraft/server/EntitySpider.java
@@ -29,7 +29,7 @@ public class EntitySpider extends EntityMonster {
 
     protected void h() {
         super.h();
-        this.datawatcher.a(16, new Byte((byte) 0));
+        this.datawatcher.a(16, Byte.valueOf((byte) 0)); // PandaSpigot - use valueOf()
     }
 
     public void t_() {
diff --git a/src/main/java/net/minecraft/server/EntityWither.java b/src/main/java/net/minecraft/server/EntityWither.java
index fb19bad3c01b03991f9dbaff6041acbccda3c8b3..2ac189aabc35657905052e35243516148816bb7c 100644
--- a/src/main/java/net/minecraft/server/EntityWither.java
+++ b/src/main/java/net/minecraft/server/EntityWither.java
@@ -48,10 +48,12 @@ public class EntityWither extends EntityMonster implements IRangedEntity {
 
     protected void h() {
         super.h();
-        this.datawatcher.a(17, new Integer(0));
-        this.datawatcher.a(18, new Integer(0));
-        this.datawatcher.a(19, new Integer(0));
-        this.datawatcher.a(20, new Integer(0));
+        // PandaSpigot start - use valueOf()
+        this.datawatcher.a(17, Integer.valueOf(0));
+        this.datawatcher.a(18, Integer.valueOf(0));
+        this.datawatcher.a(19, Integer.valueOf(0));
+        this.datawatcher.a(20, Integer.valueOf(0));
+        // PandaSpigot end
     }
 
     public void b(NBTTagCompound nbttagcompound) {
diff --git a/src/main/java/net/minecraft/server/EntityWolf.java b/src/main/java/net/minecraft/server/EntityWolf.java
index 469c87a179b452b5df06e79a6bc1d8e7fce07965..3a83f9a8e4ea49f20895d2eb18b3fb47bf4ddf91 100644
--- a/src/main/java/net/minecraft/server/EntityWolf.java
+++ b/src/main/java/net/minecraft/server/EntityWolf.java
@@ -87,9 +87,11 @@ public class EntityWolf extends EntityTameableAnimal {
 
     protected void h() {
         super.h();
-        this.datawatcher.a(18, new Float(this.getHealth()));
-        this.datawatcher.a(19, new Byte((byte) 0));
-        this.datawatcher.a(20, new Byte((byte) EnumColor.RED.getColorIndex()));
+        // PandaSpigot start - use valueOf()
+        this.datawatcher.a(18, Float.valueOf(this.getHealth()));
+        this.datawatcher.a(19, Byte.valueOf((byte) 0));
+        this.datawatcher.a(20, Byte.valueOf((byte) EnumColor.RED.getColorIndex()));
+        // PandaSpigot end
     }
 
     protected void a(BlockPosition blockposition, Block block) {
